openwisp_controller/connection/apps.py

```diff
@@ -106,12 +107,22 @@ def _is_update_in_progress(cls, device_pk):
 
     @classmethod
     def is_working_changed_receiver(
-        cls, instance, is_working, old_is_working, **kwargs
+        cls,
+        instance,
+        is_working,
+        old_is_working,
+        failure_reason,
+        old_failure_reason,
+        **kwargs,
     ):
         # if old_is_working is None, it's a new device connection which wasn't
         # used yet, so nothing is really changing and we won't notify the user
         if old_is_working is None:
             return
+        # don't send notification if error occurred due to connectivity issues
+        for ignore_reason in app_settings._IGNORE_CONNECTION_NOTIFICATION_REASONS:
```

Comment:
Wouldn't be better to have this a class attribute instead of a variable in `settings.py`?

Since we do not allow users to edit this, it can be put anywhere, so I think having it as a class attribute here makes more sense than having it in settings.py.